README.md
# NAME

Dist::Iller::Config::Author::CSSON - Dist::Iller config

<div>
    <p>
    <img src="https://img.shields.io/badge/perl-5.14+-blue.svg" alt="Requires Perl 5.14+" />
    <img src="https://img.shields.io/badge/coverage-42.7%25-red.svg" alt="coverage 42.7%" />
    <a href="https://github.com/Csson/p5-Dist-Iller-Config-Author-CSSON/actions?query=workflow%3Amakefile-test"><img src="https://img.shields.io/github/workflow/status/Csson/p5-Dist-Iller-Config-Author-CSSON/makefile-test" alt="Build status at Github" /></a>
    </p>
</div>

# VERSION

Version 0.0328, released 2020-12-28.

# SYNOPSIS

    # in iller.yaml
    +config: Author::CSSON
    splint: 1

# DESCRIPTION

Dist::Iller::Config::Author::Csson is a [Dist::Iller](https://metacpan.org/pod/Dist::Iller) configuration. The plugin list is in `share/author-csson.yaml`.

# ATTRIBUTES

## travis

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Int">Int</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>0</code></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">read/write</td>
    <td style="text-align: right; padding-right: 6px; padding-left: 6px;"><code>1</code>:</td>
    <td style="padding-left: 12px;">Include [TravisYML].</td>
</tr>
</table>

<p></p>

## travis\_perl\_min

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Int">Int</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>14</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>Minimum Perl version to test on Travis. All production releases up to (and including) 'travis_perl_max' are automatically included. Only give the minor version number (eg '14' for Perl 5.14).</p>

## travis\_perl\_max

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Int">Int</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>24</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>Maximum Perl version to test on Travis. See 'travis_perl_min'.</p>

## badges

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Bool">Bool</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>1</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>Include Badge::Depot badges or not.</p>

## filepath

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Path::Tiny#Path">Path</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>author-csson.yaml</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>Path to the plugin configuration file, relative to the installed share dir location.</p>

## homepage

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Str">Str</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>https://metacpan.org/release/[distribution_name]</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read/write</td>
</tr>
</table>

<p>URL to the distribution's homepage.</p>

## installer

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Str">Str</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>MakeMaker</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read/write</td>
</tr>
</table>

<p>The installer plugin to be used.</p>

## is\_private

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Int">Int</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>0</code></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">read/write</td>
    <td style="text-align: right; padding-right: 6px; padding-left: 6px;"><code>1</code>:</td>
    <td style="padding-left: 12px;">Include [UploadToStratopan].</td>
</tr>
</table>

<p></p>

## is\_task

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Bool">Bool</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>0</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>If set to a true value it will include [TaskWeaver] instead of [PodWeaver].</p>

## main\_module

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Str">Str</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>The package name</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>Override this attribute when there's more than one config in a distribution. It uses the main_module's sharedir location for the config files.</p>

## splint

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Int">Int</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>0</code></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">read/write</td>
    <td style="text-align: right; padding-right: 6px; padding-left: 6px;"><code>1</code>:</td>
    <td style="padding-left: 12px;">Include Pod::Elemental::Transformer::Splint in weaver.ini</td>
</tr>
</table>

<p></p>

## global

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">Maybe[<a href='https://metacpan.org/pod/Types::Standard#InstanceOf'>InstanceOf</a>['Dist::Iller::DocType::Global']]</td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional</td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p></p>

# ENVIRONMENT VARIABLES

## FAKE\_RELEASE

If set to a true value this will include \[FakeRelease\] and remove either \[UploadToCPAN\] or \[UploadToStratopan\].

# SOURCE

[https://github.com/Csson/p5-Dist-Iller-Config-Author-CSSON](https://github.com/Csson/p5-Dist-Iller-Config-Author-CSSON)

# HOMEPAGE

[https://metacpan.org/release/Dist-Iller-Config-Author-CSSON](https://metacpan.org/release/Dist-Iller-Config-Author-CSSON)

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Dist-Iller-Config-Author-Csson

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.0328 2020-12-28T12:17:24Z UTC
 [Config Changes]
 - Add Githubactions badge.
 - Remove Kwalitee and Cpantesters badges (hopefully temporarily).
 - Move PromptIfStale from 'build' to 'release'.

 [Other Changes]
 - Set required Perl version to 5.14, since that is what Dist::Zilla
   requires.

 [Stats & Dependency Changes]
 - 4 files changed, 14 insertions(+), 11 deletions(-)
 - (run req) ~ perl 5.010000 --> 5.014000
 - (dev req) + Badge::Depot::Plugin::Githubactions (any)
 - (dev req) - Badge::Depot::Plugin::Cpantesters
 - (dev req) - Badge::Depot::Plugin::Kwalitee
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0327 --> 0.0328

0.0327 2020-12-27T18:32:55Z UTC
 [Config Changes]
 - Replaced [TravisYML] with [Author::CSSON::GithubActions]
 - Temporarily (hopefully) removed [Test::Kwalitee::Extra], due to
   installation issues
 - Use $self.distribution_name in gitignore, so that it gets set
   automatically

 [Stats & Dependency Changes]
 - 5 files changed, 67 insertions(+), 16 deletions(-)
 - (dev req) +
   Dist::Zilla::Plugin::Author::CSSON::GithubActions::Workflow::TestWithMakefile
   0.0106
 - (dev req) - Dist::Zilla::Plugin::Test::Kwalitee::Extra
 - (dev sug) ~ Dist::Iller 0.1408 --> 0.1409
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0326 --> 0.0327

0.0326 2020-12-15T20:52:38Z UTC
 [Config Changes]
 - Removed .travis.yml from Author::CSSON::MapMetro
 - Disabled travis by default

 [Stats & Dependency Changes]
 - 4 files changed, 3 insertions(+), 13 deletions(-)
 - (dev req) - Badge::Depot::Plugin::Travis
 - (dev req) - Dist::Zilla::Plugin::TravisYML
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0325 --> 0.0326

0.0325 2020-06-29T19:25:28Z UTC
 [Config Changes]
 - Bump some version prereqs.

 [Stats & Dependency Changes]
 - 3 files changed, 4 insertions(+), 4 deletions(-)
 - (dev req) ~ Dist::Zilla 6.012 --> 6.015
 - (dev req) ~ Dist::Zilla::Plugin::PromptIfStale 0.055 --> 0.057
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0324 --> 0.0325

0.0324 2020-01-30T18:51:13Z UTC
 [Config Changes]
 - Bump some Dist::Zilla plugins.

 [Stats & Dependency Changes]
 - 3 files changed, 9 insertions(+), 9 deletions(-)
 - (dev req) ~ Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.008 --> 0.009
 - (dev req) ~ Dist::Zilla::Plugin::CheckChangesHasContent 0.008 --> 0.011
 - (dev req) ~ Dist::Zilla::Plugin::Git 2.045 --> 2.046
 - (dev req) ~ Dist::Zilla::Plugin::Git::Contributors 0.034 --> 0.035
 - (dev req) ~ Dist::Zilla::Plugin::RunExtraTests 0.028 --> 0.029
 - (dev req) ~ Dist::Zilla::Plugin::Test::Compile 2.057 --> 2.058
 - (dev req) ~ Dist::Zilla::Plugin::Test::Version 1.05 --> 1.09
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0322 --> 0.0324

0.0323 2019-07-28T18:43:50Z UTC
 [Config Changes]
 - Bump some prereqs.

 [Stats & Dependency Changes]
 - 2 files changed, 2 insertions(+), 2 deletions(-)

0.0322 2018-06-29T22:27:23Z UTC
 [Config Changes]
 - Bump some prereqs.

 [Stats & Dependency Changes]
 - 3 files changed, 5 insertions(+), 5 deletions(-)
 - (dev req) ~ Dist::Zilla 6.010 --> 6.012
 - (dev req) ~ Dist::Zilla::Plugin::Git 2.041 --> 2.045
 - (dev req) ~ Dist::Zilla::Plugin::Git::Contributors 0.030 --> 0.034
 - (dev req) ~ Dist::Zilla::Plugin::PromptIfStale 0.054 --> 0.055
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0321 --> 0.0322

0.0321 2018-04-22T10:24:57Z UTC
 [Config Changes]
 - Move [PromptIfStale] to build phase.

 [Stats & Dependency Changes]
 - 3 files changed, 5 insertions(+), 4 deletions(-)
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0320 --> 0.0321

0.0320 2018-02-28T19:06:16Z UTC
 [Config Changes]
 - Stop testing *prereq_matches use (due to unexpected failures)

 [Stats & Dependency Changes]
 - 4 files changed, 3 insertions(+), 4 deletions(-)
 - (dev req) + Dist::Zilla::Plugin::Test::Kwalitee::Extra v0.2.1
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0319 --> 0.0320

0.0319 2017-08-31T22:09:24Z UTC
 [Stats & Dependency Changes]
 - 3 files changed, 4 insertions(+), 4 deletions(-)
 - (dev req) ~ Dist::Zilla::Plugin::PromptIfStale 0.053 --> 0.054
 - (dev req) ~ Dist::Zilla::Plugin::Test::Compile 2.056 --> 2.057
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0318 --> 0.0319

0.0318 2017-07-31T14:05:07Z UTC
 [Config Changes]
 - Bump default max travis perl version to 5.24.
 - Bump some prereqs.

 [Stats & Dependency Changes]
 - 3 files changed, 5 insertions(+), 5 deletions(-)
 - (dev req) ~ Dist::Zilla 6.008 --> 6.010
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0316 --> 0.0318

0.0316 2017-06-27T20:21:03Z UTC
 [Config Changes]
 - Bump some prereqs.

 [Stats & Dependency Changes]
 - 3 files changed, 5 insertions(+), 4 deletions(-)
 - (run req) ~ Types::Standard 1.000005 --> 1.000006
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild 0.163040 -->
   0.170880
 - (dev req) ~ Dist::Zilla::Plugin::MetaProvides::Package 2.003001 -->
   2.004003
 - (dev req) ~ Dist::Zilla::Plugin::PromptIfStale (any) --> 0.053
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0313 --> 0.0315
 - 4 files changed, 7 insertions(+), 5 deletions(-)
 - (dev req) ~ Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.007 --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::Git::Contributors 0.029 --> 0.030
 - (dev req) ~ Dist::Zilla::Plugin::Test::ReportPrereqs 0.021 --> 0.027
 - (dev req) - Dist::Zilla::Plugin::Test::Kwalitee::Extra
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0315 --> 0.0316

0.0315 2017-04-30T17:33:10Z UTC
 [Config Changes]
 - Bump some plugin versions.

 [Stats & Dependency Changes]
 - 3 files changed, 5 insertions(+), 4 deletions(-)
 - (run req) ~ Types::Standard 1.000005 --> 1.000006
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild 0.163040 -->
   0.170880

Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.017.
use strict;
use warnings;

use 5.014000;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Dist::Iller config",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "Dist-Iller-Config-Author-CSSON",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.014000",
  "NAME" => "Dist::Iller::Config::Author::CSSON",
  "PREREQ_PM" => {
    "Dist::Iller" => "0.1401",
    "Moose" => "2.0000",
    "MooseX::AttributeDocumented" => "0.1003",
    "Path::Tiny" => "0.072",
    "Types::Path::Tiny" => "0.005",
    "Types::Standard" => "1.000006",
    "namespace::autoclean" => "0.22"
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.0329",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Iller" => "0.1401",
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => "2.0000",
  "MooseX::AttributeDocumented" => "0.1003",
  "Path::Tiny" => "0.072",
  "Test::More" => "0.96",
  "Types::Path::Tiny" => "0.005",
  "Types::Standard" => "1.000006",
  "namespace::autoclean" => "0.22"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}
